tock/api/views.py

```diff
@@ -171,6 +180,33 @@ def get_queryset(self):
             .filter(user_data__current_employee=True) \
             .order_by('last_name', 'first_name')
 
+class Submissions(generics.ListAPIView):
+    """
+    Returns a list of users and the number of timecards they have
+    submitted on time since the requested reporting period
+    """
+
+    serializer_class = SubmissionSerializer
+
+    def get_queryset(self):
+        rp_num = int(self.kwargs['num_past_reporting_periods'])
+
+        reporting_period = ReportingPeriod.get_most_recent_periods(
+            number_of_periods=rp_num,
+            number_of_results=1
+        ).get()
```

Comment:
Grabbing the earliest reporting period may be a bit simpler here if we used negative indexing to grab the `last` object in the list instead of modifying `get_most_recent_periods`. This works by first evaluating the QuerySet to a plain old `list` so we can use negative indexing. 


```suggestion
        earliest_reporting_period = list(ReportingPeriod.get_most_recent_periods(
            number_of_periods=rp_num,
        ))[-1]
```

Reply:
This is definitely simpler. I went with this suggestion and removed the logic I added to the reporting period model for `number_of_results`